ov/uplink_set.go

```diff
@@ -0,0 +1,236 @@
+	ManualLoginRedistributionState string            `json:"manualLoginRedistributionState,omitempty"` //"manualLoginRedistributionState":"NotSupported"
+	URI                            utils.Nstring     `json:"uri,omitempty"`                            // "uri": "/rest/uplink-sets/"e2f0031b-52bd-4223-9ac1-d91cb519d548",
+	Type                           string            `json:"type,omitempty"`                           // "type": "uplink-setV4",
+}
+
+type UplinkSetList struct {
+	Total       int           `json:"total,omitempty"`       // "total": 1,
+	Count       int           `json:"count,omitempty"`       // "count": 1,
+	Start       int           `json:"start,omitempty"`       // "start": 0,
+	PrevPageURI utils.Nstring `json:"prevPageUri,omitempty"` // "prevPageUri": null,
+	NextPageURI utils.Nstring `json:"nextPageUri,omitempty"` // "nextPageUri": null,
+	URI         string        `json:"uri,omitempty"`         // "uri": "/rest/uplink-sets?start=0&count=10"
+	Members     []UplinkSet   `json:"members,omitempty"`     // "members":[]
+}
+
+func (c *OVClient) GetUplinkSetByName(name string) (UplinkSet, error) {
+	var (
+		upSet UplinkSet
+	)
+	upSets, err := c.GetUplinkSets(fmt.Sprintf("name matches '%s'", name), "name:asc")
+	if upSets.Total > 0 {
+		return upSets.Members[0], err
+	} else {
+		return upSet, err
+	}
+}
+
+func (c *OVClient) GetUplinkSets(filter string, sort string) (UplinkSetList, error) {
+	var (
+		uri        = "/rest/uplink-sets"
+		q          map[string]interface{}
+		uplinkSets UplinkSetList
+	)
+	q = make(map[string]interface{})
+	if len(filter) > 0 {
+		q["filter"] = filter
+	}
+
+	if sort != "" {
+		q["sort"] = sort
+	}
+
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+	// Setup query
+	if len(q) > 0 {
+		c.SetQueryString(q)
+	}
+	data, err := c.RestAPICall(rest.GET, uri, nil)
+	if err != nil {
+		return uplinkSets, err
+	}
+
+	log.Debugf("GetUplinkSets %s", data)
+	if err := json.Unmarshal([]byte(data), &uplinkSets); err != nil {
+		return uplinkSets, err
+	}
+	return uplinkSets, nil
+}
+
+func (c *OVClient) GetUplinkSetById(id string) ([]string, error) {
+	var (
+		uri         = "/rest/uplink-sets/"
+		uplinkSetId = new([]string)
+	)
+	uri = uri + id
+	fmt.Println(uri)
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+	data, err := c.RestAPICall(rest.GET, uri, nil)
+	if err != nil {
+		return *uplinkSetId, err
+	}
+	log.Infof("GetUplinkSetId %s", data)
+	if err := json.Unmarshal([]byte(data), uplinkSetId); err != nil {
+		return *uplinkSetId, err
+	}
+	return *uplinkSetId, nil
+}
+
+func (c *OVClient) CreateUplinkSet(upSet UplinkSet) error {
+	log.Infof("Initializing creation of uplink-set for %s.", upSet.Name)
+	var (
+		uri = "/rest/uplink-sets"
+		t   *Task
+	)
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	t = t.NewProfileTask(c)
+	t.ResetTask()
+	log.Infof("REST : %s \n %+v\n", uri, upSet)
```

Comment:
Can change it back to Debug as we dont want it printed unless the user requires it.